packages/datadog-plugin-mongodb-core/src/index.js

```diff
@@ -12,6 +12,12 @@ class MongodbCorePlugin extends DatabasePlugin {
   // should be removed if one day this will be fixed
   static get peerServicePrecursors () { return [] }
 
+  constructor (tracer, config) {
+    super(tracer, config)
+    // this channel is for wrapping the callback of mongoose exec methods and handling store context, no handler needed
+    this.addBind('datadog:mongodb-core:collection:addQueue', () => {})
```

Comment:
Returning `undefined` here will make the current store `undefined` instead of restoring it as expected in `addQueue`. Instead there should be a `queue:start` and `queue:asyncStart` like for other asynchronous operations.

Reply:
Why do we need the `asyncStart`? there's only one place where this new channel is needed. But makes sense on the part where we should restore the context.